File: algorithm/src/main/java/baekjoon/p02965/Main.java
package baekjoon.p02965;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] inputs = {sc.nextInt(), sc.nextInt(), sc.nextInt()};
        System.out.println(solve(inputs));
    }

    private static int solve(int[] inputs) {
        int max = Math.max(inputs[1] - inputs[0], inputs[2] - inputs[1]);
        return max - 1;
    }

}

File: algorithm/src/main/java/baekjoon/p02965/Main.kt
package baekjoon.p02965

fun main(args: Array<String>) {
    val inputs = readLine()!!.split(" ").map { it.toInt() }
    println(solve(inputs))
}

private fun solve(inputs: List<Int>): Int {
    return (1 until inputs.size).map { inputs[it] - inputs[it - 1] }.max()!! - 1
}